src/flexible_type/flexible_type_converter.hpp

```diff
@@ -10,657 +10,899 @@
+    }
+  }
+
+  template <typename Integer>
+  static void set(Integer& dest, const flexible_type& src) {
+    dest = Integer(flex_int(src));
   }
 };
 
+////////////////////////////////////////////////////////////////////////////////
+//
+//  Vectors
+//
+////////////////////////////////////////////////////////////////////////////////
 
-/**
- * Case 3: All numeric types.
+/** Any sequence containers (vectors, lists, deques, etc.) of floating
+ *  point values.
  */
-template <typename T>
-struct flexible_type_converter<T, 
-    typename std::enable_if<std::is_arithmetic<T>::value>::type> {
-  static constexpr bool value = true;
-  T get(const flexible_type& val) {
-    if (val.get_type() != flex_type_enum::INTEGER && 
-        val.get_type() != flex_type_enum::FLOAT) {
-      std::string errormsg = 
-          std::string("Expecting a numeric type, But we got a ") 
-          + flex_type_enum_to_name(val.get_type());
-      throw(errormsg);
+template <> struct ft_converter<CVTR__FLEX_VEC_CONVERTIBLE_SEQUENCE> {
+
+  template <typename V> static constexpr bool matches() {
+    return (is_sequence_container<V>::value
+            && std::is_floating_point<typename first_nested_type<V>::type>::value);
+  }
+
+  template <typename Vector>
+  static void get(Vector& dest, const flexible_type& src) {
+    if(src.get_type() == flex_type_enum::VECTOR) {
+      const flex_vec& v = src.get<flex_vec>();
+      dest.assign(v.begin(), v.end());
+    } else if(src.get_type() == flex_type_enum::LIST) {
+      const flex_list& f = src.get<flex_list>();
+      dest.assign(f.begin(), f.end());
+    } else {
+      throw_type_conversion_error(src, "flex_vec");
     }
-    return T(val);
   }
-  flexible_type set(const T& val) {
-    return flexible_type(val);
+
+  template <typename Vector>
+  static void set(flexible_type& dest, const Vector& src) {
+    dest = flex_vec(src.begin(), src.end());
   }
 };
 
-/**
- * Case 4: std::vector<T> where T is a numeric type. Converts to flex_vec.
- * Must exclude the types already handled by case 1.
+////////////////////////////////////////////////////////////////////////////////
+
+/** flex_vec type, exact converter.  Prioritized above
+ *  flex_vec_convertible, but defined afterwards so we can call the
+ *  get method if the flexible_type isn't a flex_vec as well.
  */
-template <typename T>
-struct flexible_type_converter<std::vector<T>, 
-    typename std::enable_if<std::is_arithmetic<T>::value &&
-                            !is_flexible_type_member<std::vector<T>>::value
-                            >::type> {
-  static constexpr bool value = true;
-  std::vector<T> get(const flexible_type& val) {
-
-    if (val.get_type() != flex_type_enum::VECTOR) {
-      std::string errormsg = 
-          std::string("Expecting an array of numbers, But we got a ") 
-          + flex_type_enum_to_name(val.get_type());
-      throw(errormsg);
-    }
+template <> struct ft_converter<CVTR__FLEX_VEC_EXACT> {
```

Comment:
This file should ideally be sorted in the same order as the conversion test sequence.

Reply:
Yes.  I wasn't aware it wasn't.  Will do.